monero-oxide/io/src/lib.rs

```diff
@@ -179,11 +184,11 @@ pub fn read_scalar<R: Read>(r: &mut R) -> io::Result<Scalar> {
 /// Since this decodes an Ed25519 point, it does not check the point is in the prime-order
 /// subgroup. Torsioned points do have a canonical encoding, and only aren't canonical when
 /// considered in relation to the prime-order subgroup.
-pub fn decompress_point(bytes: [u8; 32]) -> Option<EdwardsPoint> {
-  CompressedEdwardsY(bytes)
+pub fn decompress_point(compressed: CompressedEdwardsY) -> Option<EdwardsPoint> {
```

Comment:
Isn't relying on `curve25519_dalek::CompressedEdwardsY` a massive footgun, as we expect callers to use `decompress_point` yet it provides a `decompress` method? Shouldn't we define our own `CompressedPoint` struct which matches `CompressedEdwardsY` as being a `[u8; 32]` wrapper, except its `decompress` method _is_ this function?